vm/cuttlefish/cuttlefish.go

```diff
@@ -116,15 +128,33 @@ func (inst *instance) Copy(hostSrc string) (string, error) {
 	deviceDst := filepath.Join(deviceRoot, filepath.Base(hostSrc))
 	pushCmd := fmt.Sprintf("adb push %s %s", gceDst, deviceDst)
 
-	if err := inst.runOnHost(5*time.Minute, pushCmd); err != nil {
+	if err := inst.runOnHost(5*time.Minute, pushCmd, false); err != nil {
 		return "", fmt.Errorf("error pushing to device: %s", err)
 	}
 
 	return deviceDst, nil
 }
 
 func (inst *instance) Forward(port int) (string, error) {
-	return "", fmt.Errorf("not implemented")
+	hostForward, err := inst.gceInst.Forward(port)
+	if err != nil {
+		return "", fmt.Errorf("failed to get IP/port from GCE instance: %s", err)
+	}
+	cmd := fmt.Sprintf("socat TCP-LISTEN:%d,fork TCP:%s", port, hostForward)
+	if err := inst.runOnHost(60*time.Second, cmd, true); err != nil {
```

Comment:
From what I see in the code, `runOnHost` will block even if the background flag is set to true. The function will only return if some error happened or the 24h timer expired.

Would it help if you invoked something like `nohup socat ... &` (with the appropriate parentheses/escaping)? Then the extra flag will not be needed at all.

Reply:
I agree that using 'nohup' makes more sense. I've updated the code to do this. We can just check that if/when we get an error back from runOnHost() that it's the expected timeout for this case.

Reply:
The updated code looks much better.

The `inst.runOnHost(10*time.Second, cmd)` command won't return immediately, it will always take 10 seconds. That is what you intended to do, right?

Reply:
Yes. I have to put a non-zero value here so I put 10 seconds to just to err on the safe side so that if any other problems happen establishing the connection (a port conflict, for example) we'll have time to catch it instead of returning immediately and missing it.

I've reduced it to one second, which should speed things up some. It's still working reliably with the shorter timeout.